lib/AppInfo/Application.php

```diff
@@ -24,31 +24,14 @@
 
 namespace OCA\TwoFactorEmail\AppInfo;
 
-use OCA\TwoFactorEmail\Event\StateChanged;
-use OCA\TwoFactorEmail\Listener\IListener;
-use OCA\TwoFactorEmail\Listener\StateChangeRegistryUpdater;
 use OCP\AppFramework\App;
 
 class Application extends App {
 
-	const APPNAME='twofactor_email';
+	const APP_NAME = 'twofactor_email';
 
-	public function __construct() {
-		parent::__construct(self::APPNAME);
-
-		$container = $this->getContainer();
-
-		$dispatcher = $container->getServer()->getEventDispatcher();
-		$dispatcher->addListener(StateChanged::class, function (StateChanged $event) use ($container) {
```

Comment:
why was this removed?

Reply:
since its not necessary? The provider is enabled somewhere else (in lib/Service/SetupService.php )

Reply:
If `StateChangeRegistryUpdater` is obsolete then please remove it

Reply:
it is removed?